static/js/composebox_typeahead.js

```diff
@@ -534,6 +534,21 @@ exports.get_stream_topic_data = (hacky_this) => {
     return opts;
 };
 
+function format_stream_topic_suggestion(topic) {
+    const [stream_id, topic_name] = topic.split(':');
+    // #stream_name>topic_name.
+    return `#${stream_data.maybe_get_stream_name(parseInt(stream_id, 10))}>${topic_name}`;
+}
+
+
+function match_stream_query_with_recent_topics(query) {
+    const topics = Array.from(recent_topics.get().keys());
+
+    return topics.filter(item => item.split(':').pop().startsWith(query)).
```

Comment:
Let's extract a couple locals here, so the function is a little easier to read and print-debug as needed.  Plus we're less likely to have "fake" line coverage if we avoid long one-liners.

Some suggestions:

* filtered_topics
* function compare